openmdao/core/system.py

```diff
@@ -3212,14 +3212,21 @@ def _write_table(self, var_type, var_data, hierarchical, print_arrays, out_strea
 
         # get list of var names in execution order, based on the order subsystems were setup
         var_list = []
+
         in_or_out = 'input' if var_type is 'input' else 'output'
+        real_vars = self._var_allprocs_abs_names[in_or_out]
+        disc_vars = self._var_allprocs_discrete[in_or_out]
+
         for subsys in self._subsystems_allprocs:
-            for n in self._var_allprocs_abs_names[in_or_out]:
-                if n in var_dict and n.startswith(subsys.pathname):
-                    var_list.append(n)
-            for n in self._var_allprocs_discrete[in_or_out]:
-                if n in var_dict and n.startswith(subsys.pathname):
-                    var_list.append(n)
+            # subsys.pathname will only be defined properly if a subsystem is local,
+            # but subsys.name will be properly defined.
+            path = '.'.join((self.pathname, subsys.name)) if self.pathname else subsys.name + '.'
```

Comment:
I think we need parens around `'.'.join( ... else subsys.name`.  Otherwise we're missing the trailing '.' if the first part of the `if` is True.

Reply:
already fixed in the next commit... sorry about that